search/django/weedid/views.py

```diff
@@ -91,6 +95,74 @@ def upload(request):
         )
 
 
+def upload_voc(request):
+    if not request.method == "POST":
+        return HttpResponseNotAllowed(request.method)
+    user = request.user
+    if not (user and user.is_authenticated):
+        return HttpResponseForbidden("You dont have access to proceed")
+    try:
+        voc_dir = request.POST["voc_dir"]
+        voc = request.FILES["voc"]
+        upload_dir = os.path.join(UPLOAD_DIR, str(user.id), voc_dir)
+        store_tmp_voc(voc, upload_dir)
+    except Exception as e:
+        return HttpResponseBadRequest(str(e))
```

Comment:
I think we need to log the exception (with `print_exc`) as well as returning a meaningful message like this. I think this should also be done with a decorator. Django Rest Framework would do this kind of stuff for us, if we had adopted it.

Reply:
Please log the exception